src/Mod/Arch/archobjects/base_rebar.py

```diff
@@ -0,0 +1,256 @@
+# ***************************************************************************
+# *   Copyright (c) 2020 Bernd Hahnebach <bernd@bimstatik.org>              *
+# *                                                                         *
+# *   This program is free software; you can redistribute it and/or modify  *
+# *   it under the terms of the GNU Lesser General Public License (LGPL)    *
+# *   as published by the Free Software Foundation; either version 2 of     *
+# *   the License, or (at your option) any later version.                   *
+# *   for detail see the LICENCE text file.                                 *
+# *                                                                         *
+# *   This program is distributed in the hope that it will be useful,       *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+
+__title__ = "FreeCAD base rebar object"
+__author__ = "Bernd Hahnebach"
+__url__ = "http://www.freecadweb.org"
+
+import FreeCAD
+
+import ArchComponent
+import Part
+
+if FreeCAD.GuiUp:
+    from PySide.QtCore import QT_TRANSLATE_NOOP
```

Comment:
This doesn't need to be guarded with `GuiUp`. You are using the `QT_TRANSLATE_NOOP` function anyway. So, if `GuiUp` is `False`, this won't be imported and it will cause an error in a terminal-only session.

`QtCore` can be imported regardless of whether the GUI exists or not, so this doesn't need to be guarded.